File: abc/abc123/abc123_a.cs
// https://atcoder.jp/contests/abc123/submissions/8438039

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = new int[6].Select(_ => int.Parse(Console.ReadLine())).ToArray();
            Console.WriteLine(input[4] - input[0] > input[5] ? ":(" : "Yay!");
        }
    }
}

File: abc/abc123/abc123_a.fs
// https://atcoder.jp/contests/abc123/submissions/12197102

let x = Array.zeroCreate<int> 6

for i in [ 0..5 ] do
    x.[i] <- stdin.ReadLine() |> int

if x.[4] - x.[0] > x.[5] then ":("
else "Yay!"
|> stdout.WriteLine
